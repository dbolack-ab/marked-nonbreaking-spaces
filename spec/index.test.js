import { marked } from 'marked';
import nonbreakingSpaces from '../src/index.js';

function trimLines(s) {
  return s.split('\n').map(l => l.trim()).join('\n');
}

describe('Non-Breaking Spaces', ()=>{
  test('Single Space', function() {
    marked.use(nonbreakingSpaces());
    expect(marked(trimLines(':>\n\n'))).toMatchSnapshot();
  });

  test('Double Space', function() {
    marked.use(nonbreakingSpaces());
    expect(marked(trimLines(':>>\n\n'))).toMatchSnapshot();
  });

  test('Triple Space', function() {
    marked.use(nonbreakingSpaces());
    expect(marked(trimLines(':>>>\n\n'))).toMatchSnapshot();
  });

  test('Many Space', function() {
    marked.use(nonbreakingSpaces());
    expect(marked(trimLines(':>>>>>>>>>>\n\n'))).toMatchSnapshot();
  });

  test('Multiple sets of Spaces', function() {
    marked.use(nonbreakingSpaces());
    expect(marked(trimLines(':>>>\n:>>>\n:>>>'))).toMatchSnapshot();
  });

  test('Pair of inline Spaces', function() {
    marked.use(nonbreakingSpaces());
    expect(marked(trimLines(':>>:>>'))).toMatchSnapshot();
  });

  test('Space directly between two paragraphs', function() {
    marked.use(nonbreakingSpaces());
    expect(marked(trimLines('Line 1\n:>>\nLine 2'))).toMatchSnapshot();
  });

  test('Ignored inside a code block', function() {
    marked.use(nonbreakingSpaces());
    expect(marked(trimLines('```\n\n:>\n\n```\n'))).toMatchSnapshot();
  });
});
