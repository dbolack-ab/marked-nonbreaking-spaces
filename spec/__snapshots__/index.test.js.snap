// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Non-Breaking Spaces Double Space 1`] = `
"<p>&nbsp;&nbsp;</p>
"
`;

exports[`Non-Breaking Spaces Ignored inside a code block 1`] = `
"<pre><code>
:&gt;
</code></pre>
"
`;

exports[`Non-Breaking Spaces Many Space 1`] = `
"<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</p>
"
`;

exports[`Non-Breaking Spaces Multiple sets of Spaces 1`] = `
"<p>&nbsp;&nbsp;&nbsp;
&nbsp;&nbsp;&nbsp;
&nbsp;&nbsp;&nbsp;</p>
"
`;

exports[`Non-Breaking Spaces Pair of inline Spaces 1`] = `
"<p>&nbsp;&nbsp;&nbsp;&nbsp;</p>
"
`;

exports[`Non-Breaking Spaces Single Space 1`] = `
"<p>&nbsp;</p>
"
`;

exports[`Non-Breaking Spaces Space directly between two paragraphs 1`] = `
"<p>Line 1
&nbsp;&nbsp;
Line 2</p>
"
`;

exports[`Non-Breaking Spaces Triple Space 1`] = `
"<p>&nbsp;&nbsp;&nbsp;</p>
"
`;
